File: files/CH-114.py
import pandas as pd

path = "CH-114 Merge.xlsx"

input1 = pd.read_excel(path, usecols = "B:C", nrows = 7, skiprows = 1)
input2 = pd.read_excel(path, usecols = "B:C", nrows = 5, skiprows = 12)
test = pd.read_excel(path, usecols="H:J", nrows=7, skiprows=1).rename(columns=lambda x: x.replace('.1', ''))
test["price"] = test["price"].astype(str)

input1['Product ID'] = input1['Product ID'].str.split(',')
input1 = input1.explode('Product ID')
r1 = input1.merge(input2, left_on='Product ID', right_on='product id') \
           .groupby('Date').agg({'price': lambda x: x.iloc[0].astype(str)}).reset_index()
r2 = input1.merge(r1, on='Date', how='left').fillna({'price': '-'}).groupby('Date').agg({'Product ID': ','.join, 'price': 'first'}).reset_index()

print(r2.equals(test)) # True

File: files/100-199/114/CH-114.R
library(tidyverse)
library(readxl)

path = "files/CH-114 Merge.xlsx"
input1 = read_excel(path, range = "B2:C9")
input2 = read_excel(path, range = "B13:C18")
test  = read_excel(path, range = "H2:J9")

r1 = input1 %>%
  separate_rows(`Product ID`, sep = ",") %>%
  inner_join(input2, by = c("Product ID" = "product id")) %>%
  summarise(price = first(price) %>% as.character(), .by = Date) 
r2 = input1 %>%
  left_join(r1, by = "Date") %>%
  replace_na(list(price = "-"))

identical(r2, test)
#> [1] TRUE